devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+/**
+ *  Copyright 2016 SmartThings, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  Fidure Thermostat, Based on ZigBee thermostat (SmartThings)
+ *
+ *	Author: Fidure
+ *	Date: 2014-12-13
+ *  Updated: 2017-7-19
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition (
+		name: "Fidure 2nd Generation Thermostat",
+		namespace: "smartthings/tile-ux",
+		author: "SmartThings/Fidure") {
+
+		capability "Thermostat"
+		capability "Relative Humidity Measurement"
+		
+		capability "Temperature Measurement"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Sensor"
+		capability "lock" // for locking the keypad on the device.  
+		
+		command "tempUp"
+		command "tempDown"
+		command "heatUp"
+		command "heatDown"
+		command "coolUp"
+		command "coolDown"
+		command "setTemperature", ["number"]
```

Comment:
Don't expose method that doesn't have an implementation

Reply:
removed setTemperature